### DISCUSSION TITLE
In react native, how to use cache to initialize provider?

### DISCUSSION THREAD

**Discussion by blackhu0804** [vercel/swr#2254] [Q&A]
# Bug report

## Description / Observed Behavior

I want to achieve the same effect on the web. But my method of getting localstorage is asynchronous. How can I achieve this effect please?

## Expected Behavior

https://swr.vercel.app/docs/advanced/cache#localstorage-based-persistent-cache

## Additional Context

1.3.0

**Comment by nandorojo** [comment]
You could use MMKV if you're willing to install a native library. That uses sync storage persistence with better performance than AsyncStorage. It also works on Web (I added this).

https://github.com/mrousavy/react-native-mmkv

**Comment by piotr-cz** [comment]
In my library [swr-idb-cache](https://github.com/piotr-cz/swr-idb-cache#setup):

1. Cache provider loads cached data asynchronously
1. *After* it's ready, the `<SWRConfig value={{ provider }} />` is created
